api/envoy/api/v2/core/address.proto

```diff
@@ -74,6 +74,40 @@ message TcpKeepalive {
   google.protobuf.UInt32Value keepalive_interval = 3;
 }
 
+// Generic socket option message. This would be used to set socket options that
+// might not exist in upstream kernels or precompiled Envoy binaries.
+message SocketOption {
+  // An optional name to give this socket option
+  // No special meaning is assumed.
+  string readable_name = 1;
```

Comment:
I think `name` is sufficient here. Maybe points out it's just for debug/logging (I assume?)

Reply:
But that collides with name down below which corresponds to the name argument in the sockopt calls themselves.

Reply:
Then maybe `description`?

Reply:
I'm guessing this will just be used for logging? If so, maybe log_string or something?